design-proposals/dynamic-primary-pod-nic-name.md

```diff
@@ -527,3 +531,95 @@ The primary pod interface name is set by the container runtime (for example cri-
+in the domain XML.
+
+Without intervention, a change to the primary interface name in the target pod will result in a migration failure due
+to a naming mismatch. Such a failure could, in turn, lead to an upgrade failure. 
+
+## Solution
+
+### Inferring the target pod's primary interface name
+
+All the alternatives listed bellow depend on knowing the target pod's primary interface name.
+However, the current logic in the VMI controller only provides the primary interface name of the source.
+
+To address this, an API change is proposed to enable the reporting if the target pod's primary interface name, allowing
+the necessary adjustments for migration to occur successfully: 
+
+```go
+type VirtualMachineInstanceMigrationState struct {
+...
+TargetPodPrimaryInterfaceName string `json:"targetPodPrimaryInterfaceName,omitempty"`
+...
+}
+```
+
+When the target pod is ready, the migration controller will:
+1. Identify its primary interface name (simillar to the process used by the VMI controller).
+2. Report its value using the newly added field in the `VirtualMachineInstanceMigrationState` struct.
+
+
+### Adding a hook to modify the domain in the target virt-launcher
+
+Currently, just before starting the migration, the source virt-launcher the option to modify the domain XML. However,
+the target virt-launcher does not have the ability to modify the domain XML upon receiving it from the source, as this
+process occurs between two libvirt daemons.
+
+The target virt-handler and virt-launcher will consider the target primary interface name when executing the network setup flow.
+A hook point will be introduced in the target virt-launcher, which will be invoked upon receiving the domain XML from the source
+and before the VM starts. This hook will trigger logic to adjust the primary interface if necessary.
+
+#### Pros
+- Provides greater flexibility by allowing the domain to be modified in the target, enabling dynamic adjustments during migrations.
+- Allows for future extensibility, making it easier to handle other changes as needed in the migration process.
+- This suggestion can be deferred and implemented when the need arises, as it does not depend on the source virt-launcher.
+- Integrate nicely with network binding plugins' sidecars.
+
+#### Cons
+- Requires additional integration and potentially development with libvirt to provide the hook point.
+
+## Alternatives
+### Alternative 1 - Modifying the domain XML at the source virt-launcher
+
+In this alternative, the domain XML would be modified on the source virt-launcher prior to migration. 
+The source virt-launcher would handle adjustments to the primary interface name. 
+
+The target virt-handler will consider the target primary interface name when executing the network setup flow.
+
+#### Pros
+- Could be implemented entirely within KubeVirt, without the need for additional dependencies. 
```

Comment:
This is already an established an tested process that handles CPU affinity and various disk volume readjustments - for me, this is the preferred way to proceed.

Reply:
Added this pro.